File: backend/test_api.js
const http = require('http');

const data = JSON.stringify({
  email: 'dr.mehta@vitalmind.com',
  password: 'Doctor123!'
});

const options = {
  hostname: '127.0.0.1',
  port: 5000,
  path: '/api/v1/auth/login',
  method: 'POST',
  headers: {
    'Content-Type': 'application/json',
    'Content-Length': data.length
  }
};

const req = http.request(options, res => {
  let body = '';
  res.on('data', d => body += d);
  res.on('end', () => {
    const token = JSON.parse(body).token;
    console.log('Token:', token ? 'Success' : 'Fail', body);
    
    // Now fetch patient
    const req2 = http.request({
      hostname: '127.0.0.1',
      port: 5000,
      path: '/api/v1/patients/4adb29e4-bc62-4d8d-a788-05ecc8488bd1',
      method: 'GET',
      headers: { 'Authorization': 'Bearer ' + token }
    }, res2 => {
      let body2 = '';
      res2.on('data', d => body2 += d);
      res2.on('end', () => console.log('Patient API Status:', res2.statusCode, body2));
    });
    req2.end();
  });
});

req.write(data);
req.end();

File: backend/test_api.py
import requests
import json

BASE_URL = "http://localhost:5000/api/v1"

# 1. Login
response = requests.post(f"{BASE_URL}/auth/login", json={
    "email": "john@test.com",
    "password": "Patient123!"
})

if response.status_code == 200:
    data = response.json()
    token = data.get("access_token")
    print("Login successful, token:", token[:10] + "...")
    
    # 2. Fetch doctors
    docs_response = requests.get(f"{BASE_URL}/doctors/", headers={
        "Authorization": f"Bearer {token}"
    })
    
    print("Doctors Status Code:", docs_response.status_code)
    try:
        print("Doctors Response JSON:", json.dumps(docs_response.json(), indent=2))
    except Exception as e:
        print("Failed to parse JSON:", docs_response.text)
else:
    print("Login failed:", response.status_code, response.text)
